enterprise/backend/test/metabase_enterprise/api/gsheets_test.clj

```diff
@@ -134,10 +134,9 @@
 
 (defmacro with-sample-db-as-dwh [& body]
   "We need an attached dwh for these tests, so let's have the sample db fill in for us:"
-  `(try
-     (t2/update! :model/Database :id 1 {:is_attached_dwh true})
-     ~@body
-     (finally (t2/update! :model/Database :id 1 {:is_attached_dwh false}))))
+  (let [db-sym (gensym "db-")]
```

Comment:
my changes for some reason make the tests in this namespace fail: https://github.com/metabase/metabase/actions/runs/14790039957/job/41525297547?pr=57416#step:4:310

it's probably bc test order were changed because the test in namespace expects a db with id `1` exists.

it's not a good assumption to make anyway, so I've changed to use a temporarily db. it seems the tests don't care which db its operating on, so this is actually a much better isolation.

Reply:
Only concern is test run time, maybe we could use a fixture to share the same temp db for all thes tests in this namespace.

Reply:
we're mocking an empty without any tables so I don't think it adds much run time tbh.

Reply:
do you mind measuring? i had a sense it was fairly heavy (> 100ms) but very happy to be wrong